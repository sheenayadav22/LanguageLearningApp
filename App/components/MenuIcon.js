import React, { useCallback } from 'react';
import { DrawerActions, useNavigation } from '@react-navigation/native';
import { TouchableOpacity } from 'react-native';
import { Feather as DefaultFeather } from '@expo/vector-icons';

function MenuIcon() {
  const navigation = useNavigation();

  const openDrawer = useCallback(() => {
    navigation.dispatch(DrawerActions.openDrawer());
  },[]);

  return (
    <TouchableOpacity onPress={openDrawer}>
      <DefaultFeather name="menu" size={24} style={{marginLeft: 25}} />
    </TouchableOpacity>
  );
}

export default MenuIcon;
