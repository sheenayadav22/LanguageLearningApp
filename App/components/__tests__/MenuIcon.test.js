import React from 'react';
import renderer from 'react-test-renderer';
import MenuIcon from '../MenuIcon';

jest.mock('@react-navigation/native', () => {
  const actualnav = jest.requireActual('@react-navigation/native');
  return {
    ...actualnav,
    useNavigation: () => {
      navigation: () => jest.fn()
    },
  }
});

// Mocks vector-icon Feather to prevent warnings
jest.mock('@expo/vector-icons', () => {
  const { View } = require('react-native');
  return {
    Feather: View
  };
});

it('renders correctly', () => {
  const tree = renderer.create(<MenuIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});
