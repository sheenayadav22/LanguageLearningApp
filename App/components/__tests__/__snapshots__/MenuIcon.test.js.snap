// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    name="menu"
    size={24}
    style={
      Object {
        "marginLeft": 25,
      }
    }
  />
</View>
`;
